src/npm/index.js

```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 if (process.env.NODE_ENV === 'production') {
-  module.exports = require('./recoil.production.js');
+  module.exports = require('./recoil.production');
```

Comment:
Removed the explicit extension so that the React Native bundler (Metro) imports `recoil.{mode}.native.js`. If this is too terse, we can add an `index.native.js` entry file and a `react-native` key to the package.json.

Reply:
This works; however resolution problems are harder to debug later if they go wrong :) For instance, this may resolve the browser version by mistake depending on how the react-native bundler is configured.

Can we stick with a more explicit approach as in #108? Or does that not work with the React Native bundler?